wordpress-mobile/WordPress-iOS
# Add `post_type` property on some `wpios_editor_` events

Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/13527

## Not Testable
`editor_draft_saved`
- This event currently doesn't fire and has been separated out to https://github.com/wordpress-mobile/WordPress-iOS/issues/13729. The area that handles this is [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/13708/files#diff-56d3ee28947d056476717395cb71f4d7L519-R519) and can be indirectly tested using `wpios_editor_quick_draft_saved`

## To test:

For the following events, I've included the steps to help check the analytics flow. These can be validated in Tracks or by running the app in XCode and filtering the console logs on "🔵 Tracked:"
For each event, you should expect to see the event fire with the `post_type` set to "page" for a Page and "post" for Posts.

---
`editor_post_created`

- **Create using FAB button** 
    - From your selected site. 
    - Select the floating action button at the bottom right of the screen
        - You'll be prompted to "Create New:" "Blog post" or "Site page."
    - Select either "Blog post" or "Site page."
    - The event should fire 
- **Create via the Page List** 
    - From your selected site. 
    - Select the "Pages" Action button under the site name or "Site Pages."
        - You'll be taken to the "Site Pages" screen
    - Select the ➕in the top right of the screen
    - The event should fire with `post_type` set to `page.`
- **Create via the Post List** 
    - From your selected site. 
    - Select the "Posts" Action button under the site name or "Blog Posts."
        - You'll be taken to the "Blog Posts" screen
    - Select the ➕in the top right of the screen
    - The event should fire with `post_type` set to `post.`
- **Create via Adding Media** 
    - From your selected site. 
    - Select the "Media" Action button under the site name
        - You'll be taken to the "Media" screen
    - Select the ➕in the top right of the screen and add a new media file
    - Wait for the upload to finish
        - You should see a Toast at the bottom of the screen with a button labeled "Write Post."
    - Select "Write Post" 
    - The event should fire with `post_type` set to `post.`
- **Create via the App Icon** 
    - On the device Home screen
    - Force Press or LongPress the App Icon
    - Select "New Photo Post" or "New Post" 
    - The event should fire with `post_type` set to `post.`
- **Create via Universal Links** 
    - In an outside app such as Notes
    - Create a Universal Link for a new post
        - If running a local build `wpdebug://newpost?content=Test&title=Post` will work 
        - If running an alpha build `wpalpha://newpost?content=Test&title=Post` will work 
       - **Note** content is a required field
    - The event should fire with `post_type` set to `post`

`editor_post_published`
- Create a Page or a Post
- Make some edits
- Select "Publish"
- The event should fire 

`editor_post_updated`
- **Update via Update Button** 
    - Open an existing Page or a Post
    - Make some edits 
    - Select "Update"
    - The event should fire 
- **Update via Backing out** 
    - Open an existing Page or a Post
    - Make some edits 
    - Select the "X" in the top left 
    - Select "Update Draft"
    - The event should fire 

`editor_post_scheduled`
- Create a Page or a Post
- Make some edits
- Select the `...` in the top right corner
- Select "Post Settings"
- Select "Publish" and set a date and time to the future
- Select "Status" and change to "Scheduled."
- Back out to the Page or Post
- Select "Schedule"
- The event should fire 

`editor_quick_draft_saved`
- Create a Page or a Post
- Make some edits
- Select the `...` in the top right corner
- Select "Save as Draft"
- The event should fire 

## PR submission checklist:
